src/portable/espressif/esp32s2/dcd_esp32s2.c

```diff
@@ -203,7 +213,41 @@ void dcd_init(uint8_t rhport)
                  USB_RESETDETMSK_M |
                  USB_DISCONNINTMSK_M; // host most only
 
-  dcd_connect(rhport);
+  if (did_persist) {
+    USB0.grstctl &= ~USB_TXFNUM_M;
+    USB0.grstctl |= 0x10 << USB_TXFNUM_S;
+    USB0.grstctl |= USB_TXFFLSH;
+    USB0.grxfsiz = 52;
+
+    for (int n = 0; n < USB_IN_EP_NUM; n++) {
+      USB0.in_ep_reg[n].diepint = USB_D_XFERCOMPL0_M | USB_D_TXFEMP0_M;
+      USB0.in_ep_reg[n].diepctl &= ~USB_D_STALL0_M; // clear Stall
+      USB0.in_ep_reg[n].diepctl |= USB_D_CNAK0 | USB_D_EPENA0; // clear NAK
+    }
+    USB0.dtknqr4_fifoemptymsk &= ~(0x7F);
+
+    USB0.gnptxfsiz = (16 << USB_NPTXFDEP_S) | (USB0.grxfsiz & 0x0000ffffUL);
+
+    USB0.daintmsk |= USB_OUTEPMSK0_M | USB_INEPMSK0_M;
+    USB0.doepmsk |= USB_SETUP0 | USB_XFERCOMPLMSK;
+    USB0.diepmsk |= USB_TIMEOUTMSK_M | USB_DI_XFERCOMPLMSK_M;//USB_INEPNAKEFFMSK
+
+    USB0.gintmsk |= USB_IEPINTMSK_M | USB_OEPINTMSK_M;
+    USB0.gotgint = ~0; //clear OTG ints
+    USB0.gintsts = ~0; //clear pending ints
+    enum_done_processing();
+    dcd_event_bus_signal(0, DCD_EVENT_BUS_RESET, true);
```

Comment:
the 3rd argument is `in_isr` which should be false in this case, else it will use the freeRTOS fromISR() API version.